viewto-frontend/src/models/ViewStudy.ts

```diff
@@ -33,12 +33,13 @@ export class ViewStudy {
         this.id = data.id;
         this.name = data.ViewName;
         this.sasakiId = data.ViewId;
+        let hackyIndex = 0;
         this.focuses = data.objects
             .filter((x: any) => x.speckle_type == ViewObjectTypes.context.speckle_type && x.Content_Type == "Potential")
             .map((x: any) => this._focusContextToWeb(x));
         this.obstructors = data.objects
             .filter((x: any) => x.speckle_type == ViewObjectTypes.context.speckle_type && x.Content_Type != "Potential")
-            .map((x: any) => this._obstructorContextToWeb(x));
+            .map((x: any) => this._obstructorContextToWeb(x, hackyIndex++));
```

Comment:
the hacky index could be replaced with the index: `.map((x: any, i: number) => this._obstructorContextToWeb(x, i))`